examples/basic_edit.py

```diff
@@ -44,10 +44,10 @@
 print('External links:', list(page.extlinks()))
 
 print('Uploading image')
-site.upload(open('test-image.png', 'rb'), prefix + '-test-image.png', 'desc', ignore=True)
+site.upload(open('test-image.png', 'rb'), f'{prefix}-test-image.png', 'desc', ignore=True)
 print('Uploading image for the second time')
-site.upload(open('test-image.png', 'rb'), prefix + '-test-image.png', 'desc', ignore=True)
-image = site.Images[prefix + '-test-image.png']
+site.upload(open('test-image.png', 'rb'), f'{prefix}-test-image.png', 'desc', ignore=True)
```

Comment:
while we're modernizing, I think these `open` calls ought to have an explicit `encoding` arg, right?

Reply:
That, and `open()` should probably use a context manager, so this should be something like this instead imo:

```suggestion
with open('test-image.png', 'rb', encoding='utf-8') as f:
    site.upload(f, f'{prefix}-test-image.png', 'desc', ignore=True)
```